app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        if args not in cache_dict:
```

Comment:
use (args, tuple(kwargs.items()), func.__name__) as key in cache_dict because args, kwargs and function name are all identifiers of result